torchvision/models/maxvit.py

```diff
@@ -0,0 +1,854 @@
+            nn.Tanh(),
+            nn.Linear(block_channels[-1], num_classes, bias=False),
+        )
+
+        self._init_weights()
+
+    def forward(self, x: Tensor) -> Tensor:
+        x = self.stem(x)
+        for block in self.blocks:
+            x = block(x)
+        x = self.classifier(x)
+        return x
+
+    def _init_weights(self):
+        for m in self.modules():
+            if isinstance(m, nn.Conv2d):
+                nn.init.normal_(m.weight, std=0.02)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+            elif isinstance(m, nn.BatchNorm2d):
+                nn.init.constant_(m.weight, 1)
+                nn.init.constant_(m.bias, 0)
+            elif isinstance(m, nn.Linear):
+                nn.init.normal_(m.weight, std=0.02)
+                if m.bias is not None:
+                    nn.init.zeros_(m.bias)
+
+
+def _maxvit(
+    # stem parameters
+    stem_channels: int,
+    # block parameters
+    block_channels: List[int],
+    block_layers: List[int],
+    stochastic_depth_prob: float,
+    # conv parameters
+    squeeze_ratio: float,
+    expansion_ratio: float,
+    # conv + transformer parameters
+    # normalization_fn is applied only to the conv layers
+    # activation_fn is applied both to conv and transformer layers
+    normalization_fn: Callable[..., nn.Module],
+    activation_fn: Callable[..., nn.Module],
+    # transformer parameters
+    head_dim: int,
+    mlp_ratio: int,
+    mlp_dropout: float,
+    attn_dropout: float,
+    # partitioning parameters
+    partition_size: int,
+    # Weights API
+    weights: Optional[WeightsEnum],
+    progress: bool,
+    # task parameters
+    num_classes: int = 1000,
+    # kwargs,
+    **kwargs,
+) -> MaxVit:
+
+    if weights is not None:
+        _ovewrite_named_param(kwargs, "num_classes", len(weights.meta["categories"]))
+        assert weights.meta["min_size"][0] == weights.meta["min_size"][1]
+        _ovewrite_named_param(kwargs, "input_size", weights.meta["min_size"])
+
+    input_size = kwargs.pop("input_size", (224, 224))
+    input_channels = kwargs.pop("input_channels", 3)
+
+    model = MaxVit(
+        input_channels=input_channels,
+        stem_channels=stem_channels,
+        num_classes=num_classes,
+        block_channels=block_channels,
+        block_layers=block_layers,
+        stochastic_depth_prob=stochastic_depth_prob,
+        squeeze_ratio=squeeze_ratio,
+        expansion_ratio=expansion_ratio,
+        normalization_fn=normalization_fn,
+        activation_fn=activation_fn,
+        head_dim=head_dim,
+        mlp_ratio=mlp_ratio,
+        mlp_dropout=mlp_dropout,
+        attn_dropout=attn_dropout,
+        partition_size=partition_size,
+        input_size=input_size,
+    )
+
+    if weights is not None:
+        model.load_state_dict(weights.get_state_dict(progress=progress))
+
+    return model
+
+
+_COMMON_META = {
+    "categories": _IMAGENET_CATEGORIES,
+}
```

Comment:
Nit: You don't need to define one given you only have one set of weights. You could just move the key below.